schemas/v3.1/schema.yaml

```diff
@@ -13,7 +13,7 @@ properties:
   jsonSchemaDialect:
     type: string
     format: uri
-    default: 'https://spec.openapis.org/oas/3.1/dialect/base'
+    default: 'https://spec.openapis.org/oas/3.1/dialect/base/WORK-IN-PROGRESS'
```

Comment:
Do we really need this change?

It will complicate the build script, so what is the return on investment?

Reply:
@ralfhandl I'm not sure I understand the question.  Each document can be iterated independently.  These needed to have dates from the beginning.

Reply:
I assumed it was a conscious decision to _not_ date-stamp the `dialect` and `meta` files, and not to have a `vocab` file:

~~~mermaid
flowchart LR
    schema_base
    schema
    dialect
    meta
    vocab
    schema --> dialect
    schema_base --> schema
    schema_base --> dialect
    dialect --> meta
    dialect --> vocab
    meta --> vocab
~~~

Reply:
@ralfhandl I wasn't here when that decision was made as I would never have approved it.

If these aren't dated, **_how do you plan to handle it if the dialect or vocabulary schemas need to be changed for any reason?_**

Yes, they are simpler and less likely to change.  That doesn't mean they'll never need to change- if not for 3.1 then in the future, in which case we need to have the pattern set correctly and not be stuck.  I don't know why they were done this way but as far as I'm concerned it was an error.

Again, in practice we can set them once and they probably won't change after that, so the build script needn't get much more complicated right now.  We could:
* Manually publish the dialect and vocabulary schemas with an initial dated URI
* Hardcode a dependency on that initial dated URI for now:
    * Always use those hardcoded initial dated URIs when `schema-base` is updated
    * Error out if the dialect or vocabulary schemas change, so we don't have to code support for that unlikely scenario now (or deal with triggering on multi-document interdependent changes)
    * Changes to `schema` aren't impacted by this at all

Reply:
Let's do the needful, even if it hurts 😎